src/components/VmActions/ConsoleConfirmationModal.js

```diff
@@ -92,6 +92,14 @@ export default connect(
       dispatch(setConsoleInUse({ vmId: vm.get('id'), consoleInUse: null }))
       dispatch(setConsoleLogon({ vmId: vm.get('id'), isLogon: null }))
     },
-    onDownloadConsole: ({ usbFilter, force }) => dispatch(downloadConsole({ vmId: vm.get('id'), usbFilter, consoleId, hasGuestAgent: vm.get('ssoGuestAgent'), force })),
+    onDownloadConsole: ({ usbFilter, skipSSO, userId }) => dispatch(
+      downloadConsole({
+        vmId: vm.get('id'),
+        usbFilter,
+        consoleId,
+        hasGuestAgent: vm.get('ssoGuestAgent'),
+        skipSSO: skipSSO || vm.get('sessions').find(s => s.getIn(['user', 'id']) === userId) !== undefined, // Parameter frr skiping SSO authorization
```

Comment:
Can you extract the `vm.get('sessions').find(s => s.getIn(['user', 'id']) === userId) !== undefined` check to a utility function and then reuse it in `consoleInUse()` in `console/index.js`?

I recommend a name like `doesVmSessionExistForUserId` or similar.

Reply:
+1 for readable names on utilities :)